rclcpp/test/rclcpp/test_publisher_with_type_adapter.cpp

```diff
@@ -118,8 +118,8 @@ struct TypeAdapter<std::string, rclcpp::msg::LargeMessage>
     const custom_type & source,
     ros_message_type & destination)
   {
-    destination.size = source.size();
-    std::memcpy(destination.data.data(), source.data(), source.size());
+    destination.size = std::min(source.size(), destination.data.max_size());
+    std::memcpy(destination.data.data(), source.data(), destination.size);
```

Comment:
By the way, while I agree that the old code was dangerous, I think this isn't great either.  In particular, if we don't have room in the destination to copy the source in, then we'll copy something truncated, which also seems wrong.  Particularly in this test case, I think we should throw an exception if the destination isn't large enough.